mne/beamformer/_rap_music.py

```diff
@@ -172,9 +172,19 @@ def _make_dipoles(times, poss, oris, sol, gof):
 def _compute_subcorr(G, phi_sig):
     """Compute the subspace correlation."""
     Ug, Sg, Vg = linalg.svd(G, full_matrices=False)
+    # Now we look at the actual rank of the forward fields
+    # in G and handle the fact that it might be rank defficient
+    # eg. when using MEG and a sphere model for which the
+    # radial component will be truely 0.
+    rank = np.sum(Sg > (Sg[0] * 1e-6))
+    if rank == 0:
+        return 0, np.zeros(len(G))
+    rank = max(rank, 2)  # rank cannot be 1
+    Ug, Sg, Vg = Ug[:, :rank], Sg[:rank], Vg[:rank]
     tmp = np.dot(Ug.T.conjugate(), phi_sig)
     Uc, Sc, Vc = linalg.svd(tmp, full_matrices=False)
-    X = np.dot(np.dot(Vg.T, np.diag(1. / Sg)), Uc)  # subcorr
+    Sg_pinv = 1. / Sg
+    X = np.dot(np.dot(Vg.T, np.diag(Sg_pinv)), Uc)  # subcorr
```

Comment:
... and for bonus points, if all you ever really use is `X[:, 0]` as is the case here, you should be able to speed up computation further by just dotting with `Uc[:, 0]` instead of `Uc`.